### PR TITLE
Documentation for Features In-Progress

### PR DESCRIPTION
**Pull request by smithis7**
### :pencil: Description

**Type:** :memo: `documentation`

Currently, the development roadmap section of the TARDIS documentation is highly outdated. In addition to updating it, this PR makes a place in the documentation for placing documentation for features not yet implemented into TARDIS, as this has been demonstrated to be necessary given all the new developments going on in the collaboration.

This is in accordance with discussions at TARDIS-con.

Note: I think I may be missing something about Hydrogen NLTE. Please let me know if this is the case, and if so what to put for it.

### :pushpin: Resources

See the new page: https://tardis-sn.github.io/tardis/pull/2105/contributing/in_progress/index.html

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [x] Other method (describe)
- [ ] My changes can't be tested (explain why)

Docs built on GitHub.

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [x] I updated the documentation according to my changes
- [x] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: docs/contributing/in_progress/index.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+============================
+
+Many new TARDIS developments are made through GSOC students. To learn more about potential GSOC projects, see the `GSOC ideas page <https://tardis-sn.github.io/gsoc/tardis-gsoc-ideas.html>`_.
+
+
+
+Restructure
+===========
+
+Restructuring TARDIS to be more modularized in order to facilitate the addition of new physics modules and improve maintainability. See https://github.com/orgs/tardis-sn/projects/5.
```
Comment (andrewfullard) [inline docs/contributing/in_progress/index.rst:22]: This URL is to a private project board, so you should remove it

File: docs/contributing/in_progress/index.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+New Physics
+===========
+
+TARDIS is currently capable of simulating spherically symmetric type Ia and stripped envelope supernovae at a single point in time in the earlier stages of the explosion. To expand to more types of supernovae, such as type Ib, Ic, and IIP, and to model a larger variety of times after the explosion, new physics is necessary.
```
Comment (andrewfullard) [inline docs/contributing/in_progress/index.rst:28]: "...new physics capabilities are necessary" flows better in my opinion

File: docs/contributing/in_progress/index.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+* Treatment of bound-free and free-free processes have been developed and are being integrated into TARDIS.
+
+
+Simulate Non-Homologous Expansion
+---------------------------------
+
+* Currently, TARDIS models supernovae as expanding homologously. This is not always the case in many types of supernovae, and much of the simulation needs to be modified in order to drop this assumption.
+
+
+Simulate Spherically Asymmetric Supernovae
```
Comment (andrewfullard) [inline docs/contributing/in_progress/index.rst:52]: Cut "spherically"

**Review by andrewfullard — CHANGES_REQUESTED** [review]

File: docs/contributing/in_progress/index.rst (hunk 1 above)
Comment (smithis7) [reply docs/contributing/in_progress/index.rst]: Ahh, I didn't realize it was private

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**